scrapy_zyte_api/handler.py

```diff
@@ -143,13 +142,32 @@ def _build_client(settings):
                 "Your Zyte API key is not set. Set ZYTE_API_KEY to your API key."
             )
 
+    def _load_fallback_handler(self, scheme: str) -> None:
+        if scheme == "http":
+            setting = "_ZYTE_API_FALLBACK_HTTP_HANDLER"
+        elif scheme == "https":
+            setting = "_ZYTE_API_FALLBACK_HTTPS_HANDLER"
+        else:
+            raise ValueError(f"Unsupported scheme {scheme}")
+        setting_value = self._crawler.settings.get(
+            setting, DOWNLOAD_HANDLERS_BASE[scheme]
+        )
+        dhcls = load_object(setting_value)
+        self._fallback_handler = create_instance(
+            dhcls, settings=None, crawler=self._crawler
+        )
+
     def download_request(self, request: Request, spider: Spider) -> Deferred:
         api_params = self._param_parser.parse(request)
         if api_params is not None:
             return deferred_from_coro(
                 self._download_request(api_params, request, spider)
             )
-        return self.fallback_handler.download_request(request, spider)
+        if not self._fallback_handler:
+            scheme = urlparse_cached(request).scheme
+            self._load_fallback_handler(scheme)
```

Comment:
What do you think about defining 2 handler classes, for http and https? In this case they will know which scheme they handle.

it seems we can still advice users to use a single handler for both schemas if they don't use addon, because handler only needs to know schema to use the _ZYTE_API_FALLBACK_HTTP(S)_HANDLER options, which are private.

Reply:
I guess the question is what is better :) What do you prefer?

Reply:
I think having two additional classes that only contain several lines of code is fine here, I'll try to implement that and see how it looks.

Reply:
I think it looks fine, though I needed to also make a 3rd class for the normal use case. It also no longer reads the default settings and just hardcodes the default handler, like in the current master.